queryserver-orchestrator/src/main/java/org/apache/phoenix/queryserver/orchestrator/LoggingCanarySink.java

```diff
@@ -0,0 +1,75 @@
+package org.apache.phoenix.queryserver.orchestrator;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import org.apache.phoenix.tool.CanaryTestResult;
+import org.apache.phoenix.tool.PhoenixCanaryTool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class LoggingCanarySink implements PhoenixCanaryTool.Sink {
```

Comment:
nit: please add descriptive comments for all relevant classes and methods

Reply:
yet to be addressed.